apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/model/SimpleXAPI.kt

```diff
@@ -1549,111 +1552,152 @@ object ChatController {
     }
   }
 
-  suspend fun apiReceiveFiles(rh: Long?, fileIds: List<Long>, userApprovedRelays: Boolean, encrypted: Boolean, inline: Boolean? = null): List<Long> {
-    val failures = mutableListOf<Long>()
-    val serversToApprove = mutableSetOf<String>()
-    val filesIdsFromNonApprovedSources = mutableListOf<Long>()
+  suspend fun receiveFiles(rhId: Long?, user: UserLike, fileIds: List<Long>, userApprovedRelays: Boolean = false, auto: Boolean = false) {
+    val filesAccepted = mutableListOf<AChatItem>()
+    val filesIdsToApprove = mutableStateListOf<Long>()
+    val srvsToApprove = mutableSetOf<String>()
+    val otherFileErrs = mutableListOf<CR>()
 
-    fileIds.forEach { fileId ->
-      val r = sendCmd(rh, CC.ReceiveFile(fileId, userApprovedRelays = userApprovedRelays, encrypt = encrypted, inline = inline))
+    for (fileId in fileIds) {
+      val r = sendCmd(
+        rhId, CC.ReceiveFile(
+          fileId,
+          userApprovedRelays = userApprovedRelays,
+          encrypt = appPrefs.privacyEncryptLocalFiles.get(),
+          inline = null
+        )
+      )
+      if (r is CR.RcvFileAccepted) {
+        filesAccepted.add(r.chatItem)
```

Comment:
why not to call this right here?
```kotlin
chatItemSimpleUpdate(rhId, user, it)
```

I don't see a reason to put this into list and update a lot of items in one moment (it can cause notifications to fire)